src/business-logic/services/rundown-timeline-service.ts

```diff
@@ -39,12 +40,14 @@ export class RundownTimelineService implements RundownService {
   public async activateRundown(rundownId: string): Promise<void> {
     await this.assertNoRundownIsActive()
     const rundown: Rundown = await this.rundownRepository.getRundown(rundownId)
-
+    const snapBefore = rundown.getInfinitePiecesMap()
     rundown.activate()
 
     await this.buildAndPersistTimeline(rundown)
-
-    this.rundownEventEmitter.emitInfinitePiecesUpdatedEvent(rundown)
+    const snapAfter = rundown.getInfinitePiecesMap()
+    if (isDeepStrictEqual(snapBefore, snapAfter)) {
```

Comment:
You don't need to do a deep equals for this.
We just want to know if the Pieces in the Map before are the same as the Pieces in the Map after.
So if the length differs, we know an update happened and we need to emit.
If the length doesn't differ, then we need to check if it's the same Pieces that are in the Map and for that we can just use the ids.

And example to check for ids could look like this:
![image](https://github.com/tv2/sofie-server/assets/23049647/af969413-b861-440b-b75e-b6f5226c5082)

There is some looping so it is not necessarily great, but it's only looping over small string arrays instead of doing a deep comparison.